influxdb3_write/src/last_cache.rs

```diff
@@ -381,15 +405,28 @@ impl LastCache {
             *target = LastCacheState::Store(LastCacheStore::new(
                 self.count.into(),
                 self.ttl,
-                self.schema.clone(),
+                Arc::clone(&self.schema),
+                self.series_key.as_ref(),
             ));
         }
-        target
-            .as_store_mut()
-            .expect(
-                "cache target should be the actual store after iterating through all key columns",
-            )
-            .push(row);
+        let store = target.as_store_mut().expect(
+            "cache target should be the actual store after iterating through all key columns",
+        );
+        let Some(new_columns) = store.push(row, self.accept_new, &self.key_columns) else {
+            // Unless new columns were added, and we need to update the schema, we are done.
+            return;
+        };
+
+        let mut sb = ArrowSchemaBuilder::new();
+        for f in self.schema.fields().iter() {
+            sb.push(Arc::clone(f));
+        }
+        for (name, data_type) in new_columns {
+            sb.push(Arc::new(ArrowField::new(name, data_type, true)));
+        }
+        let new_schema = Arc::new(sb.finish());
```

Comment:
Something worth noting is that the write buffer is validating the incoming writes, so we shouldn't have to worry about new fields being written with incompatible types in subsequent lines of LP. I do think ensuring that ordering discrepancies like above should be handled.

Reply:
https://github.com/influxdata/influxdb/pull/25125/commits/a129d003397bbb859a112980aee3de6286d1adcf switched to using `SchemaBuilder::try_merge` to prevent conflicts, and with the added test case, the above scenario will not cause issues.

Reply:
https://github.com/influxdata/influxdb/pull/25125/commits/2917fc149766ac349a4cb297f86ea7dfb6395797 Looks to have resolved this.